rocAL/include/augmentations/geometry_augmentations/node_crop_resize.h

```diff
@@ -31,15 +31,24 @@ class CropResizeNode : public CropNode {
     CropResizeNode() = delete;
     void init(float area, float aspect_ratio, float x_center_drift, float y_center_drift);
     void init(FloatParam *area, FloatParam *aspect_ratio, FloatParam *x_drift_factor, FloatParam *y_drift_factor);
+    void init(unsigned int crop_h, unsigned int crop_w, float x_drift, float y_drift,
+              RocalResizeInterpolationType interpolation_type = RocalResizeInterpolationType::ROCAL_LINEAR_INTERPOLATION);
     unsigned int get_dst_width() { return _outputs[0]->info().max_shape()[0]; }
     unsigned int get_dst_height() { return _outputs[0]->info().max_shape()[1]; }
-    std::shared_ptr<RocalRandomCropParam> get_crop_param() { return _crop_param; }
+    std::shared_ptr<CropParam> get_crop_param() { 
+        if(_is_random_crop) 
+            return _crop_param; 
+        else 
+            return _crop_fixed_param; }
 
    protected:
     void create_node() override;
     void update_node() override;
 
    private:
     std::shared_ptr<RocalRandomCropParam> _crop_param;
```

Comment:
Can we not add Fixed crop support in RocalRandomCropParam? 
What is the difference between RocalRandomCropParam and RocalCropParam?

Reply:
RocalRandomCrop uses aspect ratio and area factor for cropping whereas RocalCropParam uses width and height for cropping. Since we have to support both crop params here, we create two params in the node and set them accordingly in the init functions

Reply:
Please add it as a comment.

Reply:
I'm returning either crop_param or crop_fixed_param. Since both had CropParam as base class, it can be sent as CropParam shared pointer here